### ISSUE TITLE
How to serialize EncryptedValue ?

### ISSUE THREAD

**Issue by sagara11** [IronCoreLabs/recrypt-rs#180]

**Comment by skeet70** [comment]
Duplicate of #168 

We still haven't prioritized this, since the values are `pub` and can be serialized that way. A PR adding serde to types behind a `serde` feature flag would be welcomed though.

**Comment by sagara11** [comment]
@skeet70 okay but how do i suppose to get the SignedValue struct to pub because it is in the internal crate and it keeps telling me that the internal crate is private and i just cannot get it to public. Do you have any idea about that? Please help!

**Comment by sagara11** [comment]
The reason why i need to get the SignedValue struct is EncryptedValue is derived from this struct

**Comment by sagara11** [comment]
they keep warning me this although i have changed the scope of internal module in recrypt.rs file 
![image](https://github.com/IronCoreLabs/recrypt-rs/assets/50282671/ac6f7639-a2f0-4cb3-ab0a-50a310d4130a)

**Comment by skeet70** [comment]
> The reason why i need to get the SignedValue struct is EncryptedValue is derived from this struct

I'm not sure I understand why you need to do this. Could you clarify the use case a bit more?

**Comment by sagara11** [comment]
@skeet70 hmmm well just skip all of these. In conclusion, do you have any recommendation or idea for serializing the Encrypted data struct :( i have been struggling with these days

**Comment by sagara11** [comment]
From my point of view, i need to transfer the EncryptedValue to the other system to save or shared or sth like that and then i can build up again this struct in the ironcore system use functionalities like transform function

**Comment by skeet70** [comment]
Yeah, the [link from the answer to the issue I marked this as a duplicate of](https://github.com/IronCoreLabs/ironoxide/blob/main/src/internal/document_api.rs#L871-L885) is how _we_ serialize EncryptedValue to another system. That code is from one of our other libraries that's using recrypt's public API just like you're trying to do.

**Comment by sagara11** [comment]
Okay let me try researching it thank you for your help!
